### PR TITLE
remove tune.py and ray dependency so torch can be updated

### LINKED ISSUE #408: Upgrade PyTorch
We're currently using Pytorch 1.7.1 but the current release is 1.10. It'd be nice to have access to the new PyTorch profiler, which wasn't introduced until 1.8.1:
https://pytorch.org/blog/introducing-pytorch-profiler-the-new-and-improved-performance-tool/

### PR DESCRIPTION
**Pull request by ismael-mendoza**
Closes #408 

- [x] Updated pytorch 
- [x] updated pytorch lightning
- [x] fix optimizer_closure error
- [x] remove warnings from torch / pytorch lightning that have been piling up

### CODE REVIEW

**Comment by jeff-regier** [comment]
Looks like you're getting some errors from Ray. But I don't think we need that dependency any more, right? Perhaps this is a good time to get rid of it.

**Comment by ismael-mendoza** [comment]
@jeff-regier - we haven't used ray in a while but the `bliss/tune.py` still exists which uses ray and that is how we got some of the hyper-parameters for optimizing m2.

**Comment by ismael-mendoza** [comment]
It seems to be a known issue:  https://github.com/ray-project/ray/issues/20741

**Comment by jeff-regier** [comment]
I thought no one was using `tune.py`. If so, it seems better to remove it from master.

File: bliss/models/galaxy_net.py [hunk 1]
```diff
@@ -209,6 +209,9 @@ def optimizer_step(
         if optimizer_idx == 1:
             if self.trainer.global_step > self.residual_delay_n_steps:
                 optimizer.step(closure=optimizer_closure)
+            else:
+                # call closure by itself to run `training_step` + `backward` without optimizer step
+                optimizer_closure()
```
Comment (dereklhansen) [inline bliss/models/galaxy_net.py:214]: I'm a little confused by this change. Why would we want to run the `training_step` and `backward` on the residual autoencoder before we start optimizing it?

File: pyproject.toml [hunk 2]
```diff
@@ -61,15 +59,10 @@ addopts = "-ra"
...
+  "ignore:numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 232 from PyObject:RuntimeWarning",
   "ignore:numpy.ndarray size changed:RuntimeWarning",
-  "ignore:`pytorch_lightning.metrics.*` module has been renamed to `torchmetrics.*` and split off to its own package:DeprecationWarning",
-  "ignore:Using or importing the ABCs from 'collections' instead of from 'collections.abc':DeprecationWarning",
-  "ignore:The `LightningModule.datamodule` property is deprecated in v1.3 and will be removed in v1.5. Access the datamodule through using `self.trainer.datamodule` instead:DeprecationWarning",
-  "ignore:ray.services.get_node_ip_address has been moved to ray.util.get_node_ip_address. 'ray.services' will be removed after Ray 1.4.:DeprecationWarning",
+  "ignore:.*torch.meshgrid.*:UserWarning",
 ]
 markers = [
   "multi_gpu: marks tests that requires multiple gpus to run",
```
Comment (jeff-regier) [inline pyproject.toml:68]: Is this marker still relevant? I think tune.py was the only code using multiple GPUs.

Comment (ismael-mendoza) [reply pyproject.toml:68]: yeah let's get rid of it

**Comment by ismael-mendoza** [comment]
@jeff-regier or @dereklhansen have any of you seen the current message in the CI ? It's strange because I don't get this in the server when I run `pytest`

**Comment by jeff-regier** [comment]
Maybe relevant: https://stackoverflow.com/questions/70520120/attributeerror-module-setuptools-distutils-has-no-attribute-version

**Comment by ismael-mendoza** [comment]
same error pinning version 1.9 of pytorch, I might put off upgrading till later

**Comment by jeff-regier** [comment]
The post suggests that pinning `setuptools==59.5.0` will work. It's strange, the issue doesn't come up for me either on the deeplearning-01 server

**Comment by ismael-mendoza** [comment]
I pinned `torch` to `>= 1.8, < 1.9` and the test pass now but I get these warnings only in the CI. Should we merge @jeff-regier  ? 

``` 
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/torch/utils/tensorboard/__init__.py:3
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/torch/utils/tensorboard/__init__.py:3
  /home/runner/.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/torch/utils/tensorboard/__init__.py:3: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if not hasattr(tensorboard, '__version__') or LooseVersion(tensorboard.__version__) < LooseVersion('1.15'):
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/astropy/version.py:26
  /home/runner/.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/astropy/version.py:26: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    for j, piece in enumerate(LooseVersion(version).version[:3]):
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/astropy/utils/introspection.py:161
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/astropy/utils/introspection.py:161
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/astropy/utils/introspection.py:161
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/astropy/utils/introspection.py:161
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/astropy/utils/introspection.py:161
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/astropy/utils/introspection.py:161
  /home/runner/.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/astropy/utils/introspection.py:161: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    return LooseVersion(have_version) >= LooseVersion(version)
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/seaborn/rcmod.py:82
  /home/runner/.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/seaborn/rcmod.py:82: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(mpl.__version__) >= "3.0":
../../../.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/setuptools/_distutils/version.py:351
  /home/runner/.cache/pypoetry/virtualenvs/bliss-UE3m6-Ul-py3.8/lib/python3.8/site-packages/setuptools/_distutils/version.py:351: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    other = LooseVersion(other)
-- Docs: https://docs.pytest.org/en/stable/warnings.html
```

**Comment by jeff-regier** [comment]
Yes, seems fine. We can figure out how to update to pytorch all the way another time.

**Comment by amogkam** [comment]
Hey @ismael-mendoza @jeff-regier fyi newer versions of Ray Tune now support PTL 1.5, if this was the blocker for using Ray Tune.

**Pull request merged**